src/object/cli_obj.c

```diff
@@ -3401,6 +3401,45 @@ obj_shard_list_comp_cb(struct shard_auxi_args *shard_auxi,
 
 	shard_arg = container_of(shard_auxi, struct shard_list_args, la_auxi);
 	if (obj_auxi->req_tgts.ort_grp_size == 1) {
+		if (obj_is_ec(obj_auxi->obj) &&
+		    obj_auxi->opc == DAOS_OBJ_RECX_RPC_ENUMERATE &&
+		    shard_arg->la_recxs != NULL) {
+			int		i;
+			daos_obj_list_t	*obj_args;
+
+			obj_args = dc_task_get_args(obj_auxi->obj_task);
+			for (i = 0; i < shard_arg->la_nr; i++) {
+				int index;
+
+				index = obj_args->incr_order?i:(shard_arg->la_nr - 1 - i);
+
+				if (shard_arg->la_recxs[index].rx_idx & PARITY_INDICATOR)
+					obj_recx_parity_to_daos(obj_get_oca(obj_auxi->obj),
+								&shard_arg->la_recxs[index]);
+
+				rc = merge_recx(iter_arg->merged_list,
+						shard_arg->la_recxs[index].rx_idx,
+						shard_arg->la_recxs[index].rx_nr, 0);
+				if (rc)
+					return rc;
+			}
+
+			if (! obj_args->incr_order) {
+				/* Reverse the output merged list */
+				d_list_t	*it = iter_arg->merged_list;
+				d_list_t	*tmp = NULL;
+
+				while (tmp == NULL || it != iter_arg->merged_list) {
+					tmp = it->next;
+					it->next = it->prev;
+					it->prev = tmp;
+					it = tmp;
+				}
+			}
```

Comment:
hmm, instead of changing here, it seems you should use d_list_for_each_entry_reverse_safe() in dump_key_and_anchor_eof_check(), which dump the recxs from merge list to output buffer.

Reply:
- [ ] Do not change list but dump it on reverse order

Reply:
At the end, the function `dump_key_and_anchor_eof_check()` seems to not be the best suited for inverting the merged list and the function `d_list_for_each_entry_reverse_safe()`. Thus, I added this last function to gurt (and an associated unit test) and inverted the list when it was copied into the `obj_list_recxs_cb()` function.